src/kademlia/find_data.cpp

```diff
@@ -110,10 +109,12 @@ void find_data::start()
 void find_data::got_write_token(node_id const& n, std::string write_token)
 {
 #ifndef TORRENT_DISABLE_LOGGING
-	get_node().observer()->log(dht_logger::traversal
-		, "[%p] adding write token '%s' under id '%s'"
-		, static_cast<void*>(this), aux::to_hex(write_token).c_str()
-		, aux::to_hex(n).c_str());
+	auto logger = get_node().observer();
+	if (logger != nullptr && logger->should_log(dht_logger::traversal))
```

Comment:
I think it would be clearer to add the { and } here, since the statement is multiple lines